packages/cli/src/cms/commands/__tests__/push-status.test.ts

```diff
@@ -207,6 +238,272 @@ describe('handlePushStatus()', () => {
+
+      remotes.getPush.mockResolvedValueOnce({
+        ...pushResponseStub,
+        isMainBranch: true,
+        status: {
+          preview: {
+            deploy: { status: 'success', url: 'https://preview-test-url' },
+            scorecard: [],
+          },
+          production: {
+            deploy: { status: 'pending', url: 'https://production-test-url' },
+            scorecard: [],
+          },
+        },
+      });
+
+      remotes.getPush.mockResolvedValueOnce({
+        ...pushResponseStub,
+        isMainBranch: true,
+        status: {
+          preview: {
+            deploy: { status: 'success', url: 'https://preview-test-url' },
+            scorecard: [],
+          },
+          production: {
+            deploy: { status: 'running', url: 'https://production-test-url' },
+            scorecard: [],
+          },
+        },
+      });
+
+      remotes.getPush.mockResolvedValueOnce({
+        ...pushResponseStub,
+        isMainBranch: true,
+        status: {
+          preview: {
+            deploy: { status: 'success', url: 'https://preview-test-url' },
+            scorecard: [],
+          },
+          production: {
+            deploy: { status: 'success', url: 'https://production-test-url' },
+            scorecard: [],
+          },
+        },
+      });
+
+      const result = await handlePushStatus(
+        {
+          domain: 'test-domain',
+          organization: 'test-org',
+          project: 'test-project',
+          pushId: 'test-push-id',
+          'max-execution-time': 1000,
+          wait: true,
+        },
+        mockConfig
+      );
+
+      expect(result).toEqual({
+        preview: {
+          status: 'success',
+          url: 'https://preview-test-url',
+          scorecard: [],
+          isOutdated: false,
+          noChanges: false,
+        },
+        production: {
+          status: 'success',
+          url: 'https://production-test-url',
+          scorecard: [],
+          isOutdated: false,
+          noChanges: false,
+        },
+      });
+    }, 30000);
+  });
+
+  describe('"ignore-deployment-failures" option', () => {
+    it('should throw error if option is false', async () => {
+      process.env.REDOCLY_AUTHORIZATION = 'test-api-key';
+
+      remotes.getPush.mockResolvedValueOnce({
+        ...pushResponseStub,
+        status: {
+          preview: {
+            deploy: { status: 'failed', url: 'https://preview-test-url' },
+            scorecard: [],
+          },
+        },
+      });
+
+      let caughtError;
+      try {
+        await handlePushStatus(
+          {
+            domain: 'test-domain',
+            organization: 'test-org',
+            project: 'test-project',
+            pushId: 'test-push-id',
+            'max-execution-time': 1000,
+            'ignore-deployment-failures': false,
+          },
+          mockConfig
+        );
+      } catch (error) {
+        caughtError = error;
+      }
+
+      expect(caughtError).toBeDefined();
```

Comment:
Wouldn't it be easier (and more readable) to use [toThrow](https://jestjs.io/docs/expect#tothrowerror) instead?